### PR TITLE
rebind the links when classifying

### PR DESCRIPTION
**Pull request by HannesBuseyne**
I don't seem to get why the help pane classifies some links and doesn't classify others. e.g.:
image
when this code is parsed :
```
 <tr>
    <td>URL</td>
    <td><a href='https://github.com/tudo-r/BatchJobs' target='_blank'>https://github.com/tudo-r/BatchJobs</a></td>
 </tr>
```
image
from this code:
```
<tr>
   <td>URL</td>
   <td><a href='https://github.com/mani2012/BatchQC' target='_blank'>https://github.com/mani2012/BatchQC</a></td>
</tr>
```
It's not an issue, as it's easy to reset the target when classifying the links in the async loader. But I thought it was pretty weird, thus maybe worth mentioning

### CODE REVIEW

**Comment by HannesBuseyne** [comment]
@ludov04 : problem not yet solved on first load (Rstudio processes it after all our javascript has loaded, so do not merge this yet please

**Comment by HannesBuseyne** [comment]
solved by rebinding all external links to `browseURL` rstudio requests

**Comment by ludov04** [comment]
I think help pane add target=_blank to all absolute link and add /help/... to all relative links

**Comment by HannesBuseyne** [comment]
In this specific case the (absolute) link to github was rebound to http://127.0.0.1:'help-panel_port'/... it's really weird why it rebinds that specific absolute link. But i just rebind all links to `browseURL` now before the help panel gets a chance of changing anything

**Comment by ludov04** [comment]
A more clean solution but that would require a bit of refactor would be:
- When requesting the page from RStudio, we serve the layout page but with an empty content.
- When add a data-url parameter to the <div id="#content"> that is the url of the actual page.
- From async-loader we load the page with window.replacePage by giving him the url from the data-url of #content.
- All links will be ok since we get the html by ajax

It would even solve the problem you had with the history since it will push the page to the history ;p.

You can try that if you have time :)

**Comment by HannesBuseyne** [comment]
seems much cleaner indeed :p I'll do the refactoring on this branch then if that's ok

**Comment by ludov04** [comment]
@HannesBuseyne ping me when this is ready

**Comment by HannesBuseyne** [comment]
@ludov04 : this is tested and seems to work fine. 
Most of the viewer_pane specific code is in the async_loader now (except for some datacamp light js). If you have any improvements on the refactoring/code style, feel free to tell them :)

File: assets/js/dependencies/async_loader.js [hunk 1]
```diff
@@ -2,315 +2,324 @@
...
-              }
-              rerenderBody(data,rebind, urlWParams);
-            })
-          })
-          .fail(function(error) {console.log(error.responseJSON); });
-        }
+    /*
+    bind all forms to ajax requests
+    */
+    rebind('form', 'submit', function(event) {
```
Comment (ludov04) [inline assets/js/dependencies/async_loader.js]: Nice that you extracted that into a separate function

**Pull request merged**